File: 661/Main.java
import java.io.*;
import java.util.*;

public class Main {
   public static void main(String args[]) throws IOException {
      int sequence = 1;
      String line;
      StringBuilder answer = new StringBuilder();
      BufferedReader in = new BufferedReader(new InputStreamReader(System.in));

      while ((line = in.readLine()) != null) {
         StringTokenizer tokenizer = new StringTokenizer(line);
         int n = Integer.parseInt(tokenizer.nextToken());
         int m = Integer.parseInt(tokenizer.nextToken());
         int c = Integer.parseInt(tokenizer.nextToken());
         if (n == 0 && m == 0 && c == 0) break;

         int max = 0, consumed = 0;
         int[] consumption = new int[n + 1];
         boolean blown = false;
         boolean[] device = new boolean[n + 1];

         for (int i = 1; i <= n; i++) {
            consumption[i] = Integer.parseInt(in.readLine());
            device[i] = false;
         }

         while (m-- > 0) {
            int i = Integer.parseInt(in.readLine());
            device[i] = !device[i];
            if (device[i]) {
               consumed += consumption[i];
               if (consumed > max) max = consumed;
               if (consumed > c) blown = true;
            } else {
               consumed -= consumption[i];
            }
         }
         answer.append("Sequence " + Integer.toString(sequence) + "\n");
         if (blown) {
            answer.append("Fuse was blown.\n\n");
         } else {
            answer.append("Fuse was not blown.\n");
            answer.append("Maximal power consumption was " + Integer.toString(max) + " amperes.\n\n");
         }
         ++sequence;
      }
      System.out.print(answer);
   }
}

File: 661/main.c
#include <stdio.h>

int main(int argc, char* argv[])
{
   int max, consumed;
   int i, n, m, c;
   int sequence = 1;
   int consumption[21];
   char blown;
   char device[21];

   while (scanf("%d%d%d", &n, &m, &c) == 3) {
      if (n == 0 && m == 0 && c == 0) break;
      for (i = 1; i <= n; i++) {
         scanf("%d", &consumption[i]);
         device[i] = 0;
      }
      blown = 0;
      max = consumed = 0;
      while (m-- > 0) {
         scanf("%d", &i);
         device[i] = !device[i];
         if (device[i]) {
            consumed += consumption[i];
            if (consumed > max) max = consumed;
            if (consumed > c) blown = 1;
         } else {
            consumed -= consumption[i];
         }
      }
      printf("Sequence %d\n", sequence);
      if (blown) {
         printf("Fuse was blown.\n\n");
      } else {
         printf("Fuse was not blown.\n");
         printf("Maximal power consumption was %d amperes.\n\n", max);
      }
      ++sequence;
   }

   return 0;
}
